jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/GitEnricher.java

```diff
@@ -101,58 +81,80 @@ public void create(PlatformMode platformMode, KubernetesListBuilder builder) {
-                serviceBuilder.editMetadata().addToAnnotations(getAnnotations()).endMetadata();
+                serviceBuilder.editMetadata().addToAnnotations(addAnnotations(platformMode)).endMetadata();
             }
         });
 
         builder.accept(new TypedVisitor<DeploymentBuilder>() {
             @Override
             public void visit(DeploymentBuilder builder) {
-                builder.editMetadata().addToAnnotations(getAnnotations()).endMetadata();
+                builder.editMetadata().addToAnnotations(addAnnotations(platformMode)).endMetadata();
             }
         });
 
         builder.accept(new TypedVisitor<DeploymentConfigBuilder>() {
             @Override
             public void visit(DeploymentConfigBuilder builder) {
-                builder.editMetadata().addToAnnotations(getAnnotations()).endMetadata();
+                builder.editMetadata().addToAnnotations(addAnnotations(platformMode)).endMetadata();
             }
         });
 
         builder.accept(new TypedVisitor<ReplicaSetBuilder>() {
             @Override
             public void visit(ReplicaSetBuilder builder) {
-                builder.editMetadata().addToAnnotations(getAnnotations()).endMetadata();
+                builder.editMetadata().addToAnnotations(addAnnotations(platformMode)).endMetadata();
             }
         });
 
         builder.accept(new TypedVisitor<ReplicationControllerBuilder>() {
             @Override
             public void visit(ReplicationControllerBuilder builder) {
-                builder.editMetadata().addToAnnotations(getAnnotations()).endMetadata();
+                builder.editMetadata().addToAnnotations(addAnnotations(platformMode)).endMetadata();
             }
         });
 
         builder.accept(new TypedVisitor<DaemonSetBuilder>() {
             @Override
             public void visit(DaemonSetBuilder builder) {
-                builder.editMetadata().addToAnnotations(getAnnotations()).endMetadata();
+                builder.editMetadata().addToAnnotations(addAnnotations(platformMode)).endMetadata();
             }
         });
 
         builder.accept(new TypedVisitor<StatefulSetBuilder>() {
             @Override
             public void visit(StatefulSetBuilder builder) {
-                builder.editMetadata().addToAnnotations(getAnnotations()).endMetadata();
+                builder.editMetadata().addToAnnotations(addAnnotations(platformMode)).endMetadata();
             }
         });
 
         builder.accept(new TypedVisitor<JobBuilder>() {
             @Override
             public void visit(JobBuilder builder) {
-                builder.editMetadata().addToAnnotations(getAnnotations()).endMetadata();
+                builder.editMetadata().addToAnnotations(addAnnotations(platformMode)).endMetadata();
             }
         });
     }
+
+    protected void addAnnotations(PlatformMode platformMode, Map<String, String> annotations, String gitRemoteUrl, String branch, String commitId) {
```

Comment:
It would be better if the function was pure and instead of creating side effects in the input annotations variable you returned a new Map with the annotations. This map can then be appended (if applicable) to the existing variable using `putAll`.